backend/user_service/controllers/userController.js

```diff
@@ -0,0 +1,55 @@
+const { db, auth } = require("../config/firebaseConfig.js");
+const admin = require('firebase-admin');
+
+// Function to create a new user and add them to the Firestore collection
+const addToUserCollection = async (req, res) => {
+    const { uid, email, displayName } = req.body;
+
+    try {
+        console.log("Attempting to add user to Firestore:", uid, email, displayName);
+
+        // Add the user entry to Firestore
+        await db.collection("users").doc(uid).set({
+            uid: uid,
+            email: email,
+            displayName: displayName || 'No Name',
+            isAdmin: false,
+        });
+
+        console.log("User successfully added to Firestore");
+        res.status(201).json({ message: "User created successfully", uid });
+    } catch (error) {
+        console.error("Error writing to Firestore:", error);
+        res.status(500).send("Error adding user to Firestore");
+    }
+};
+
+const checkAdminStatus = async (req, res) => {
+    const token = req.headers.authorization?.split('Bearer ')[1];
+
+    if (!token) {
+        return res.status(401).json({ success: false, message: 'Unauthorized' });
+    }
```

Comment:
isnt the authentication is already done at the `callUserFunction`

Reply:
`callUserFunction` takes the token and passes it to the backend for verification. this line in `checkAdminStatus` just makes sure that there is a token passed into the backend

Reply:
The idea is correct (verifying token), perhaps we can reduce code duplication by directly utilising the authenticate function in the middleware folder. 

We can do this by making use of middleware rather than directly calling the function here. Go to the userRoutes and add in the authenticateToken before it passes to here.